core/src/main/scala/org/apache/spark/deploy/SparkHadoopUtil.scala

```diff
@@ -130,6 +132,20 @@ class SparkHadoopUtil extends Logging {
     UserGroupInformation.loginUserFromKeytab(principalName, keytabFilename)
   }
 
+  def updateCurrentUserHDFSDelegationToken(): Unit = {
+    val conf = new Configuration()
+    val nsId = DFSUtil.getNamenodeNameServiceId(conf)
+    val isHaEnabled = HAUtil.isHAEnabled(conf, nsId)
+
+    if(isHaEnabled){
+      val ugi = UserGroupInformation.getCurrentUser
+      val uri = FileSystem.getDefaultUri(conf)
+      val map = DFSUtil.getHaNnRpcAddresses(conf)
```

Comment:
looking at `AMDelegationTokenRenewer`, theres a more fundamental issue with HDFS which spark is trying to work around. It should be checking for tokens being expired and issuing new ones, so avoiding the whole hoop-jumping exercise in `writeNewTokensToHDFS()`. Is there a JIRA for that issue? As that's clearly a problem broader than just spark. Get it fixed & backported to 2.7.x (maybe, 2.6.x) and the problem goes away.